cmd/zpool/zpool_main.c

```diff
@@ -2173,19 +2173,84 @@ show_import(nvlist_t *config)
 	}
 }
 
+typedef long long unsigned int llu;
+
+static void
+extract_timestamp_and_txg(nvlist_t *config, uint64_t *txg,
+    uint64_t *timestamp)
+{
+	verify(nvlist_lookup_uint64(config, ZPOOL_CONFIG_TIMESTAMP,
+	    timestamp) == 0);
+	verify(nvlist_lookup_uint64(config, ZPOOL_CONFIG_FOUND_TXG,
+	    txg) == 0);
+}
+
+static int
+check_pool_for_activity(libzfs_handle_t *hdl, nvlist_t *config)
+{
+	nvlist_t *nvl = NULL;
+	uint64_t original_txg, original_timestamp;
+	int rc = 0;
+	char *poolname;
+	int seconds_between_checks = 1;
+	int times_to_check = 2;
+
+	verify(nvlist_lookup_string(config, ZPOOL_CONFIG_POOL_NAME,
+	    &poolname) == 0);
+
+	extract_timestamp_and_txg(config, &original_txg, &original_timestamp);
+
+	printf("verifying pool %s is not in use; %d seconds before import.\n",
+	    poolname, times_to_check * seconds_between_checks);
+
+	for (int i = 0; i < times_to_check; i++) {
+		uint64_t txg, timestamp;
+		sleep(seconds_between_checks);
+
+		nvl = refresh_config(hdl, config);
+		if (nvl == NULL) {
+			printf("pool %s config update failed\n", poolname);
+			rc = ENOENT;
+			break;
+		}
+
+		extract_timestamp_and_txg(config, &txg, &timestamp);
+
+		if ((txg > original_txg) || (timestamp > original_timestamp)) {
+			printf("pool %s changed during import: "
+			    "original_txg %llu new_txg %llu "
+			    "original_timestamp %llu new_timestamp %llu\n",
```

Comment:
I could add the last txg to that message, but I don't think that would normally be useful to the user.